pymeasure/instruments/keysight/keysightN7776C.py

```diff
@@ -0,0 +1,256 @@
+    trigger_in = Instrument.control('TRIG0:INP?','TRIG0:INP %s',
+                                    """ Sets the incoming trigger response and arms the module. """,
+                                    validator=strict_discrete_set,
+                                    values=['IGN','NEXT','SWS'])
+
+    wavelength = Instrument.control('sour0:wav?','sour0:wav %fnm',
+                                    """ Absolute wavelength of the output light (in nanometers)""",
+                                    validator=strict_range,
+                                    values=WL_RANGE,
+                                    get_process=lambda v: v*1e9)
+
+    sweep_wl_start = Instrument.control('sour0:wav:swe:star?','sour0:wav:swe:star %fnm',
+                                    """ Start Wavelength (in nanometers) for a sweep.""",
+                                    validator=strict_range,
+                                    values=WL_RANGE,
+                                    get_process=lambda v: v*1e9)
+    sweep_wl_stop = Instrument.control('sour0:wav:swe:stop?','sour0:wav:swe:stop %fnm',
+                                    """ End Wavelength (in nanometers) for a sweep.""",
+                                    validator=strict_range,
+                                    values=WL_RANGE,
+                                    get_process=lambda v: v*1e9)
+
+    sweep_step = Instrument.control('sour0:wav:swe:step?','sour0:wav:swe:step %fnm',
+                                    """ Step width of th[e sweep (in nanometers).""",
+                                    validator=strict_range,
+                                    values=[0.0001,WL_RANGE[1]-WL_RANGE[0]])
+    sweep_speed = Instrument.control('sour0:wav:swe:speed?','sour0:wav:swe:speed %fnm/s',
+                                    """ Speed of the sweep (in nanometers per second).""",
+                                    validator=strict_discrete_set,
+                                    values=[0.5,1,50,80,200],
+                                    get_process=lambda v: v*1e9)
+    sweep_mode = Instrument.control('sour0:wav:swe:mode?','sour0:wav:swe:mode %s',
+                                    """ Sweep mode of the swept laser source """,
+                                    validator=strict_discrete_set,
+                                    values=['STEP','MAN','CONT'])
+    sweep_twoway = Instrument.control('sour0:wav:swe:rep?','sour0:wav:swe:rep %s',
+                                    """Sets the repeat mode. Applies in stepped,continuous and manual sweep mode.""",
+                                    validator=strict_discrete_set,
+                                    values=[True,False],
+                                    set_process=lambda v: ['ONEW','TWOW'][int(v)],
+                                    get_process=lambda v: bool(['ONEW','TWOW'].index(v)))
+    _sweep_check = Instrument.measurement('sour0:wav:swe:chec?',
```

Comment:
More expressive name `sweep_params_status`? (also why private, this seems useful for a user?)

Reply:
Maybe rename to `_sweep_params_consistent`?

Reply:
Renamed as suggested, I chose to make it private because I'd want a user to use `check_sweep_params` function as this handles warnings and errors